single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/styles/iidm/LimitHighlightStyleProvider.java

```diff
@@ -0,0 +1,89 @@
+package com.powsybl.sld.svg.styles.iidm;
+
+import com.powsybl.iidm.network.*;
+import com.powsybl.sld.library.ComponentLibrary;
+import com.powsybl.sld.model.graphs.Graph;
+import com.powsybl.sld.model.graphs.VoltageLevelGraph;
+import com.powsybl.sld.model.nodes.*;
+import com.powsybl.sld.model.nodes.feeders.FeederWithSides;
+import com.powsybl.sld.svg.styles.EmptyStyleProvider;
+import java.util.*;
+
+import static com.powsybl.sld.svg.styles.StyleClassConstants.*;
+
+/**
+ * @author Jamal KHEYYAD {@literal <jamal.kheyyad at rte-international.com>}
+ */
+public class LimitHighlightStyleProvider extends EmptyStyleProvider {
+    Network network;
+
+    public LimitHighlightStyleProvider(Network network) {
+        this.network = network;
+    }
+
+    @Override
+    public List<String> getEdgeStyles(Graph graph, Edge edge) {
+        Optional<String> overloadStyle = getOverloadStyle(edge);
+        return overloadStyle.map(Collections::singletonList).orElse(Collections.emptyList());
+    }
+
+    private Optional<String> getOverloadStyle(Edge edge) {
+        List<Node> nodes = edge.getNodes();
+        for (Node node : nodes) {
+            if (node instanceof FeederNode feederNode) {
+                return getHighlightFeederStateStyle(feederNode);
+            } else if (node instanceof ConnectivityNode connectivityNode && hasOverloadedAdjacentNode(connectivityNode)) {
+                return Optional.of(OVERLOAD_STYLE_CLASS);
+            }
+        }
+        return Optional.empty();
+    }
+
+    boolean hasOverloadedAdjacentNode(ConnectivityNode connectivityNode) {
+        return connectivityNode.getAdjacentNodes().stream().anyMatch(node -> {
```

Comment:
Use filter on the stream instead of that code block
```java
return connectivityNode.getAdjacentNodes().stream()
    .filter(FeederNode.class::isInstance)
    .anyMatch(node -> isOverloaded((FeederNode) feederNode));
```

Reply:
done